apps/nouns-camp/src/components/proposal-screen.js

```diff
@@ -184,54 +185,58 @@ const ProposalMainSection = ({ proposalId, scrollContainerRef }) => {
       ? formActionOverride
       : defaultFormAction;
 
-  const [pendingComment, setPendingComment] = React.useState("");
-  const [pendingSupport, setPendingSupport] = React.useState(() => {
-    if (isFinalOrSucceededState) return 2; // No signal
-    return null;
-  });
-
   const [
-    pendingRepliesByTargetFeedItemId,
-    setPendingRepliesByTargetFeedItemId,
-  ] = React.useState(() => {
-    const initialReplyTargetId = searchParams.get("reply-target");
-    if (initialReplyTargetId == null) return {};
-    return { [initialReplyTargetId]: "" };
-  });
-  const setReply = React.useCallback((targetFeedItemId, reply) => {
-    setPendingRepliesByTargetFeedItemId((s) => ({
-      ...s,
-      [targetFeedItemId]: reply,
-    }));
-  }, []);
-  const [pendingReplyTargetFeedItemIds, setPendingReplyTargetFeedItemIds] =
-    React.useState(() => Object.keys(pendingRepliesByTargetFeedItemId));
-  const [pendingRepostTargetFeedItemIds, setPendingRepostTargetFeedItemIds] =
-    React.useState(() => {
-      const initialRepostTargetId = searchParams.get("repost-target");
-      if (initialRepostTargetId == null) return [];
-      return [initialRepostTargetId];
-    });
+    {
+      comment: pendingComment,
+      support: pendingSupport,
+      replies: pendingReplies,
+      reposts: pendingReposts,
+    },
+    {
+      setComment: setPendingComment,
+      setSupport: setPendingSupport,
+      addReply,
+      setReply,
+      deleteReply,
+      addRepost,
+      deleteRepost,
+      clearPost,
+    },
+  ] = useCachedPost(`vwr:p:${proposalId}`, { searchParams });
+
+  // for finalized props, default to 'no signal' comments
+  React.useEffect(() => {
+    if (isFinalOrSucceededState && pendingComment && pendingSupport === null)
+      setPendingSupport(2);
```

Comment:
Didn’t realize this’d be more complex when cached, good catch! What’s the reason for checking pendingComment here?

Reply:
if i remove the `pendingComment` check it'll create a cached object immediately, because of the pending support. otherwise, only when something is typed will it create a cached object